examples/middleware/src/auth.rs

```diff
@@ -0,0 +1,39 @@
+use axum::{
+    extract::{FromRef, State},
+    headers::{authorization::Bearer, Authorization},
+    http::{Request, StatusCode},
+    middleware::Next,
+    response::{IntoResponse, Response},
+    TypedHeader,
+};
+use axum_jwks::Jwks;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+struct Claims {}
+
+#[derive(Clone)]
+pub struct AppState {
+    pub jwks: Jwks,
+}
+
+impl FromRef<AppState> for Jwks {
+    fn from_ref(state: &AppState) -> Self {
+        state.jwks.clone()
+    }
+}
+
+pub async fn my_middleware<B>(
```

Comment:
Could you add a docstring to this middleware? I think it's especially important for examples to be very clear about what they're doing. Something stating that it's a middleware that verifies a valid token is provided, but does not utilize any claims.

Reply:
Thanks, also rename to `validate_token`